### PR TITLE
feat: [IDP-721]: PMS SDK Integration

### PR DESCRIPTION
**Pull request by vigneswara-propelo**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/49896)

### CODE REVIEW

**Review by devesh-harness — DISMISSED** [review]

**Comment by vigneswara-propelo** [comment]
trigger uts

**Comment by vigneswara-propelo** [comment]
trigger uts

**Comment by vigneswara-propelo** [comment]
trigger uts

**Comment by vigneswara-propelo** [comment]
trigger ut0

File: dockerization/pipeline-service/scripts/replace_configs.sh [hunk 1]
```diff
@@ -162,6 +162,22 @@ if [[ "" != "$STO_MANAGER_SERVICE_SECRET" ]]; then
   export STO_MANAGER_SERVICE_SECRET; yq -i '.yamlSchemaClientConfig.yamlSchemaHttpClientMap.sto.secret=env(STO_MANAGER_SERVICE_SECRET)' $CONFIG_FILE
 fi
 
+if [[ "" != "$IDP_SERVICE_BASE_URL" ]]; then
+  export IDP_SERVICE_BASE_URL; yq -i '.yamlSchemaClientConfig.yamlSchemaHttpClientMap.idp.serviceHttpClientConfig.baseUrl=env(IDP_SERVICE_BASE_URL)' $CONFIG_FILE
+fi
+
+if [[ "" != "$IDP_SERVICE_CONNECT_TIMEOUT_IN_SECONDS" ]]; then
```
Comment (vikyathharekal) [inline dockerization/pipeline-service/scripts/replace_configs.sh:169]: Have we added these timeout ones in the manifest? Didn't see them. Maybe I might have missed them

File: pipeline-service/config/config.yml [hunk 2]
```diff
@@ -146,6 +149,12 @@ yamlSchemaClientConfig:
         connectTimeOutSeconds: 15
         readTimeOutSeconds: 15
       secret: "IC04LYMBf1lDP5oeY4hupxd4HJhLmN6azUku3xEbeE3SUx5G3ZYzhbiwVtK4i7AmqyU9OZkwB4v8E9qM"
+    idp:
+      serviceHttpClientConfig:
+        baseUrl: http://localhost:12003/
+        connectTimeOutSeconds: 15
+        readTimeOutSeconds: 15
+      secret: "IC04LYMBf1lDP5oeY4hupxd4HJhLmN6azUku3xEbeE3SUx5G3ZYzhbiwVtK4i7AmqyU9OZkwB4v8E9qM"
```
Comment (vikyathharekal) [inline pipeline-service/config/config.yml:157]: We can add this to .gitleaksignore file

File: dockerization/pipeline-service/scripts/replace_configs.sh (hunk 1 above)
Comment (vigneswara-propelo) [reply dockerization/pipeline-service/scripts/replace_configs.sh:169]: yes, I missed to add it in the manifest. Thanks for checking
I have added now. Please check https://github.com/wings-software/ng-prod-manifests/pull/3368/commits/f6d145f2575041c5c01e778819330cc24a85cb67

File: pipeline-service/config/config.yml (hunk 2 above)
Comment (vigneswara-propelo) [reply pipeline-service/config/config.yml:157]: done

**Comment by vigneswara-propelo** [comment]
trigger uts

**Comment by vigneswara-propelo** [comment]
trigger ut2

**Comment by sahilhindwani-harness** [comment]
*[`dockerization/pipeline-service/scripts/replace_configs.sh` line 169 at r5](https://reviewable.io/reviews/harness/harness-core/49896#gh-1250202968:-NZQLWkYE8GifLX_KM5e:b-h60pnf) ([raw file](https://github.com/harness/harness-core/blob/e1359d41d44c5bada4388fb455d6e8d711162f7e/dockerization/pipeline-service/scripts/replace_configs.sh#L169)):*
<details><summary><i>Previously, vigneswara-propelo (Vigneswara) wrote…</i></summary><blockquote>

yes, I missed to add it in the manifest. Thanks for checking
I have added now. Please check https://github.com/wings-software/ng-prod-manifests/pull/3368/commits/f6d145f2575041c5c01e778819330cc24a85cb67 
</blockquote></details>

This change might not be required. We can use environment variables directly in config now. 
Example: `corePoolSize: ${PIPELINE_SETUP_USAGE_CREATION_EXECUTOR_POOL_CORE_SIZE:-1}`

This would reduce dependency on making changes in replace_configs.sh

**Comment by sahilhindwani-harness** [comment]
*[`idp-service/src/main/java/io/harness/idp/pipeline/step/StepSpecTypeConstants.java` line 10 at r7](https://reviewable.io/reviews/harness/harness-core/49896#-NZ_IT-b1_J39WL-L-Rn:-NZ_IT-b1_J39WL-L-Ro:b-pr4oya) ([raw file](https://github.com/harness/harness-core/blob/d3c2fa4fc723b4572bb419c9ca51c183b1e89b9c/idp-service/src/main/java/io/harness/idp/pipeline/step/StepSpecTypeConstants.java#L10)):*
> ```Java
> package io.harness.idp.pipeline.step;
> 
> public interface StepSpecTypeConstants {
> ```

use from CI only

**Review by sahilhindwani-harness — COMMENTED** [review]
> Reviewed 15 of 24 files at r1, 7 of 7 files at r3, 1 of 5 files at r4, 2 of 2 files at r6, 5 of 5 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49896)* status: 29 file(s) to review. 2. unresolved discussions. 0 of 2 approvals obtained.

**Review by archit-harness — APPROVED** [review]
> Reviewed 15 of 24 files at r1, 7 of 7 files at r3, 1 of 5 files at r4, 1 of 2 files at r6, 5 of 5 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49896)* status: 2. unresolved discussions. 1 of 2 approvals obtained.

**Comment by vigneswara-propelo** [comment]
trigger uts

**Review by archit-harness — APPROVED** [review]
> Reviewed 10 of 12 files at r8, 2 of 2 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49896)* status: 12 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Comment by vigneswara-propelo** [comment]
trigger ut1, ut4, ut7

**Review by vigneswara-propelo — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49896)* status: 15 file(s) to review. 3. unresolved discussions. 0 of 2 approvals obtained. Waiting on team(s): del-team. 

___
*[`dockerization/pipeline-service/scripts/replace_configs.sh` line 169 at r5](https://reviewable.io/reviews/harness/harness-core/49896#gh-1250202968:-N_Z2vXg2J27JWUKU9sx:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/e1359d41d44c5bada4388fb455d6e8d711162f7e/dockerization/pipeline-service/scripts/replace_configs.sh#L169)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

This change might not be required. We can use environment variables directly in config now. 
Example: `corePoolSize: ${PIPELINE_SETUP_USAGE_CREATION_EXECUTOR_POOL_CORE_SIZE:-1}`

This would reduce dependency on making changes in replace_configs.sh

</blockquote></details>

Done
___
*[`idp-service/src/main/java/io/harness/idp/pipeline/step/StepSpecTypeConstants.java` line 10 at r7](https://reviewable.io/reviews/harness/harness-core/49896#-NZ_IT-b1_J39WL-L-Rn:-N_YzwZy9g5IcXEALZvK:bb6un1j) ([raw file](https://github.com/harness/harness-core/blob/d3c2fa4fc723b4572bb419c9ca51c183b1e89b9c/idp-service/src/main/java/io/harness/idp/pipeline/step/StepSpecTypeConstants.java#L10)):*
<details><summary><i>Previously, sahilhindwani-harness (Sahil Hindwani) wrote…</i></summary><blockquote>

use from CI only
</blockquote></details>

Done

**Comment by vigneswara-propelo** [comment]
trigger uts

**Review by archit-harness — DISMISSED** [review]
> Reviewed 7 of 7 files at r10, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49896)* status: 15 file(s) to review. 3. unresolved discussions. 1 of 2 approvals obtained. Waiting on team(s): del-team.

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 1 of 12 files at r8.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49896)* status: 15 file(s) to review. 2. unresolved discussions. Waiting on team(s): del-team.

**Review by vikyathharekal — APPROVED** [review]

**Comment by vigneswara-propelo** [comment]
trigger review

**Comment by vigneswara-propelo** [comment]
trigger uts

**Comment by vigneswara-propelo** [comment]
trigger ut0, ut2, ut3

**Review by vikyathharekal — APPROVED** [review]

**Review by archit-harness — APPROVED** [review]
> Reviewed 5 of 6 files at r11, 1 of 1 files at r12, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/49896)* status: 16 file(s) to review. 2. unresolved discussions. Waiting on team(s): del-team.

**Comment by vigneswara-propelo** [comment]
trigger uts

**Comment by vigneswara-propelo** [comment]
trigger review

**Comment by vigneswara-propelo** [comment]
trigger ut1, ut2

**Comment by vigneswara-propelo** [comment]
trigger ut1

**Pull request merged**